src/puyapy/awst_build/eb/_expect.py

```diff
@@ -68,6 +67,15 @@ def defaulter(msg: str, location: SourceLocation) -> InstanceBuilder:  # noqa: A
     return defaulter
 
 
+def not_the_type_of(node: NodeBuilder, default: Callable[[str, SourceLocation], _T]) -> _T:
```

Comment:
The name of this method sounds like it is checking the node is not a particular type (particular in the context of `expect.not_the_type_of`, but rather it is used to emit an error after that check has already been done. I think the function name should indicate that e.g. `log_incorrect_type` ?

Reply:
Changed to `not_this_type` per discussion.